its_hub/algorithms/particle_gibbs.py

```diff
@@ -210,6 +239,15 @@ def infer(
                 # duplicate the particles
                 particles = [p.deepcopy() for p in particles]
 
+                # Truncate reference particles if they were resampled as non-reference
```

Comment:
this only needs to be done for `resampled_particles` part. 
so technically it can be moved ahead before concatenating `ref_particles`